aws-lc-rs/src/agreement.rs

```diff
@@ -627,10 +744,42 @@ fn x25519_diffie_hellman<'a>(
     Ok(&buffer[0..X25519_SHARED_KEY_LEN])
 }
 
+#[inline]
+fn x25519_diffie_hellman_x509(
+    buffer: &mut [u8],
+    evp_priv_key: &LcPtr<EVP_PKEY>,
+    peer_pub_key: &[u8],
+) -> Result<(), ()> {
+    let mut priv_key = [0u8; X25519_PRIVATE_KEY_LEN];
+    if 1 != unsafe {
+        EVP_PKEY_get_raw_private_key(
+            **evp_priv_key,
+            priv_key.as_mut_ptr(),
+            &priv_key.len() as *const _ as *mut usize,
```

Comment:
Similar to above: "len is updated with the number of bytes actually written" (https://www.openssl.org/docs/man3.1/man3/EVP_PKEY_get_raw_private_key.html)

```suggestion
            &mut priv_key.len() as *const _ as *mut usize,
```

Reply:
`x25519_diffie_hellman_x509` is now gone.  (Or will be once I update the PR)